.github/workflows/update-wp-compatibility.yml

```
@@ -0,0 +1,91 @@
+name: Bump WordPress Compatibility Version
+
+on:
+  workflow_dispatch:
+    inputs:
+      plugin:
+        type: string
+        description: 'Plugin slug (leave empty for all plugins)'
+        required: false
+
+jobs:
+  prepare-matrix:
+    name: Prepare plugins matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.plugins }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Configure plugin matrix
+        id: set-matrix
+        run: |
+          PLUGINS=$(jq -r '.plugins' plugins.json)
+          if [[ -n "${{ inputs.plugin }}" ]]; then
+            SLUG="${{ inputs.plugin }}"
+            if echo $PLUGINS | jq -e '.[] | select(. == "'$SLUG'")' > /dev/null; then
+              PLUGINS="[ \"$SLUG\" ]"
+            else
+              echo "::error::Plugin '$SLUG' not found in plugins.json"
+              exit 1
+            fi
+          fi
+          echo "::notice::Updating plugins: $(echo ${PLUGINS[@]})"
+          echo "plugins=$(echo $PLUGINS | jq -c .)" >> $GITHUB_OUTPUT
+
+  update-compatibility:
+    name: Update ${{ matrix.plugin }} compatibility
+    needs: prepare-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        plugin: ${{ fromJSON(needs.prepare-matrix.outputs.matrix) }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Compare versions
+        id: version-compare
+        run: |
+          # Get WordPress.org version.
+          curl -sSL --retry 3 --retry-delay 5 --retry-all-errors --fail -o /tmp/wp-org-readme.txt "https://plugins.svn.wordpress.org/${{ matrix.plugin }}/trunk/readme.txt"
+          if [ $? -ne 0 ]; then
+            echo "::error::Could not fetch readme.txt from WordPress.org for ${{ matrix.plugin }}"
+            exit 1
+          fi
+
+          WP_ORG_VERSION=$(grep -E "^Tested up to:" /tmp/wp-org-readme.txt | sed -E 's/Tested up to:[[:space:]]*([0-9.]+)/\1/')
+          REPO_VERSION=$(grep -E "^Tested up to:" ./plugins/${{ matrix.plugin }}/readme.txt | sed -E 's/Tested up to:[[:space:]]*([0-9.]+)/\1/')
```

How about this to make it a bit easier to read:

```suggestion
          WP_ORG_VERSION=$(grep -E "^Tested up to:" /tmp/wp-org-readme.txt | awk -F ': +' '{print $2}')
          REPO_VERSION=$(grep -E "^Tested up to:" ./plugins/${{ matrix.plugin }}/readme.txt | awk -F ': +' '{print $2}')
```

Also in the case of `awk` not matching the pattern, it will result in an empty string, which would be the error case.